src/main/java/Pac.java

```diff
@@ -0,0 +1,229 @@
+                        try {
+                            unmarkTask(Integer.parseInt(inputArray[1]) - 1);
+                        } catch (NumberFormatException e) {
+                            System.out.println(newline + "\nSorry! MARK should be followed by a integer.\n"
+                                    + newline + "\n");
+                        } catch (IndexOutOfBoundsException e) {
+                            System.out.println(newline + "\nSorry! Please mention a valid task number.\n"
+                                    + newline + "\n");
+                        }
+                        break;
+                    case TODO:
+                        if (inputArray.length == 1) {
+                            throw new PacException("Please mention a task.");
+                        }
+                        addToDo(inputArray[1]);
+                        break;
+                    case DEADLINE:
+                        if (inputArray.length == 1) {
+                            throw new PacException("Please mention a task.");
+                        }
+                        try {
+                            addDeadline(inputArray[1]);
+                        } catch (ArrayIndexOutOfBoundsException e) {
+                            System.out.println(newline + "\nSorry! Command format is incorrect.\n"
+                                    + newline + "\n");
+                        }
+                        break;
+                    case EVENT:
+                        if (inputArray.length == 1) {
+                            throw new PacException("Please mention a task.");
+                        }
+                        try {
+                            addEvent(inputArray[1]);
+                        } catch (ArrayIndexOutOfBoundsException e) {
+                            System.out.println(newline + "\nSorry! Command format is incorrect.\n"
+                                    + newline + "\n");
+                        }
+                        break;
+                    case DELETE:
+                        if (inputArray.length == 1) {
+                            throw new PacException("DELETE should be followed by a integer.");
+                        }
+                        try {
+                            deleteTask(Integer.parseInt(inputArray[1]) - 1);
+                        } catch (NumberFormatException e) {
+                            System.out.println(newline + "\nSorry! MARK should be followed by a integer.\n"
+                                    + newline + "\n");
+                        } catch (IndexOutOfBoundsException e) {
+                            System.out.println(newline + "\nSorry! Please mention a valid task number.\n"
+                                    + newline + "\n");
+                        }
+                        break;
+                    case ERROR:
+                        throw new PacException("I don't know what this means");
+                }
```

Comment:
can consider rewriting these lines of code into separate functions to avoid deep nesting